components/images/LazyImage.vue

```diff
@@ -0,0 +1,207 @@
+<template>
```

Comment:
This entire file feels very "non-Vue", if that makes sense.

I feel like stuff should be moved to methods or the component should become a directive or something. As-is, it doesn't feel like a Vue component but more like the old web client.

An interesting note is that you can use `this.$el` to access HTMLElements from your code, as well as `this.$ref`.

In any case, it feels like it needs a refactoring.

Reply:
Yeah, completely agree. However, there is one problem with making this "Vue-friendly": creating the intersection observer inside the Vue component creates one observer per element. I followed something along the lines of [this](https://medium.com/@sebassegros/using-intersectionobserver-with-vue-js-8d5dd0134606) and, although it didn't had any noticeable performance hit, I doubt multiple observers are better than one for all. 

For reference also, the v-intersect of vuetify didn't work right and has the same problem, as we discussed in Matrix.

In fact, when I followed the approach like the mentioned in the link above, I thought at first I was only using one observer. But, by chance, while messing with JS console, it's when I discovered that I had a few hundreds of observers.

I discovered that, basically, every object or instance you create inside Vue context will be done by each DOM node, not one by component. I discovered this while messing with workers in blurhash and now with the observer.

Same problem applies with the scrolling measure logic (and v-scroll is also affected). I can only add the listener outside the scope of the component so there's one instance only. I did it in this LazyImage component because it was easier to me while testing, as both the scrolling detection and observer are related feature-wise. I guess the best place for this is behind plugins?

I also need some kind of input in how I can make the new window.isScrollingFast available and reactive inside Vue context (so, whenever the original one is changed, it can be watched and used inside components). Right now it's only used in LazyImage but my plans are to expand some of this to the LazyCard component as well so, at some point, I'll need this. 

I tried passing the variable to ``data()`` but doesn't work. I wonder why, as variables are references to objects, not copies, like in Python.

Reply:
I could see a Vuex store to handle all of that. Giving you access to the observer, its state, its entries and all that.

It's essentially a big variables database anyway, so that's kind of what its made for.